### ISSUE TITLE
SameDiff/libnd4j: BatchNorm needs some work for training

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#7166]
There's a few issues here:
1. No gradient/backprop op for fused_batch_norm op: https://github.com/deeplearning4j/deeplearning4j/blob/master/libnd4j/include/ops/declarable/generic/nn/fusedBatchNorm.cpp
2. Currently importing TF fused_batch_norm can have is_training=false as a fixed attribute. This is problematic as we can't then actually "unfreeze" the variables and train the model (as fused_batch_norm op will always be in test mode)
3. We have 2 other batch norm op implementations: batch_norm and batchnorm_new... IMO we should only keep one (probably the latter with mkl-dnn support)
4. No backprop implementation for fused_batch_norm, batchnorm_new
5. batchnorm_bp is wrong: mean and variance gradients aren't typically calculated in batch norm; instead, the global mean/variance estimate is generally updated during forward pass when set to training mode...

**Comment by agibsonccc** [comment]
Closing — SameDiff has been completely rewritten with a new DSP (Dynamic Shape Plan) execution engine, native plan compilation, CUDA graph replay, and Triton compilation. The architecture referenced in this issue no longer exists. If you still experience this issue with the upcoming release, please open a new issue.
